core/workflow-core/src/main/scala/edu/uci/ics/amber/core/storage/model/VirtualDocument.scala

```diff
@@ -116,4 +116,10 @@ abstract class VirtualDocument[T] extends ReadonlyVirtualDocument[T] {
     */
   def getTableStatistics: Map[String, Map[String, Any]] =
     throw new NotImplementedError("getTableStatistics method is not implemented")
+
+  /**
+    * Expire snapshots if the document supports it.
+    */
+  def expireSnapshots(): Unit =
```

Comment:
This method should not be at VirtualDocument.

You can define a trait, called `OnIceberg`, and defile this method.

At the place you want to call this method, you can convert the document into `OnIceberg` type and call it